OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignalPrefs.java

```diff
@@ -132,113 +132,52 @@
             super(name);
         }
 
+        @Override
+        protected void onLooperPrepared() {
+            super.onLooperPrepared();
+
+            // Getting handler here as onLooperPrepared guarantees getLooper() will not return null
+            mHandler = new Handler(getLooper());
+            scheduleFlushToDiskJob();
+        }
+
         private synchronized void startDelayedWrite() {
             // A Context is required to write,
             //   if not available now later OneSignal.setContext will call this again.
             if (OneSignal.appContext == null)
                 return;
 
-            if (mHandler == null) {
-                startThread();
-                mHandler = new Handler(getLooper());
-            }
+            startThread();
+            scheduleFlushToDiskJob();
```

Comment:
do you still need this call, cant you depend only on onLooperPrepared?

Reply:
`onLooperPrepared` will take care of the first flush. For 2nd, 3rd, etc calls we have to call `scheduleFlushToDiskJob` again to setup a new delayed write.

Since you had this question is there anything I can change to make this less confusing? Maybe just a comment here?

Reply:
@Jeasmine Fixed!